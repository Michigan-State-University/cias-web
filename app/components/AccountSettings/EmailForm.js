import React, { Fragment, useState, useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Formik } from 'formik';
import * as Yup from 'yup';

import Button from 'components/Button';
import Column from 'components/Column';
import FormikInput from 'components/FormikInput';
import H3 from 'components/H3';
import Modal from 'components/Modal';
import ErrorAlert from 'components/ErrorAlert';
import Row from 'components/Row';

import messages from './messages';

const validationSchema = formatMessage =>
  Yup.object().shape({
    email: Yup.string().required(formatMessage(messages.emailRequired)),
    passwordConfirmation: Yup.string().required(
      formatMessage(messages.oldPasswordRequired),
    ),
  });

const initialValues = user => ({
  email: user.email,
  passwordConfirmation: '',
});

const EmailForm = ({
  formatMessage,
  user,
  changeEmail,
  error,
  loading,
  changeErrorValue,
}) => {
  const previousLoadingState = useRef(loading);
  const [modalVisible, setModalVisible] = useState(false);
  const openModal = () => setModalVisible(true);
  const closeModal = () => setModalVisible(false);

  const onSubmit = ({ email, passwordConfirmation }, { setSubmitting }) => {
    changeEmail({ newEmail: email, oldPassword: passwordConfirmation });
    setSubmitting(false);
  };

  const handleBlur = email => () => {
    if (user.email !== email) openModal();
  };

  const handleClose = () => {
    changeErrorValue('changeEmailError', null);
    closeModal();
  };

  useEffect(() => {
    if (previousLoadingState.current && !loading && !error) handleClose();
    previousLoadingState.current = loading;
  }, [loading]);

  return (
    <Fragment>
      <Formik
        validationSchema={validationSchema(formatMessage)}
        initialValues={initialValues(user)}
        onSubmit={onSubmit}
      >
        {({ handleSubmit, values }) => {
          const inputProps = {
            height: 46,
            width: '100%',
          };
          return (
            <Fragment>
              <Modal
                visible={modalVisible}
                title={formatMessage(messages.changeYourEmail)}
                onClose={handleClose}
                maxWidth={500}
              >
                <Column>
                  <H3 mb={30}>
                    <FormattedMessage {...messages.modalDescriptipn} />
                  </H3>
                  <FormikInput
                    formikKey="passwordConfirmation"
                    placeholder={formatMessage(messages.password)}
                    label={formatMessage(messages.passwordLabel)}
                    type="password"
                    inputProps={inputProps}
                  />
                  <Row width="100%" mt={30}>
                    <Button
                      mr={20}
                      inverted
                      hoverable
                      onClick={handleClose}
                      type="button"
                    >
                      <FormattedMessage {...messages.cancel} />
                    </Button>
                    <Button
                      hoverable
                      onClick={handleSubmit}
                      type="button"
                      loading={loading}
                    >
                      <FormattedMessage {...messages.changeEmail} />
                    </Button>
                  </Row>
                  {error && <ErrorAlert mt={25} errorText={error} />}
                </Column>
              </Modal>
              <Row width="100%">
                <FormikInput
                  formikKey="email"
                  placeholder={formatMessage(messages.email)}
                  label={formatMessage(messages.emailLabel)}
                  type="text"
                  inputProps={{
                    onBlur: handleBlur(values.email),
                    ...inputProps,
                  }}
                />
              </Row>
            </Fragment>
          );
        }}
      </Formik>
    </Fragment>
  );
};

EmailForm.propTypes = {
  formatMessage: PropTypes.func,
  user: PropTypes.shape({
    email: PropTypes.string,
  }),
  loading: PropTypes.bool,
  error: PropTypes.string,
  changeEmail: PropTypes.func,
  changeErrorValue: PropTypes.func,
};

export default EmailForm;
