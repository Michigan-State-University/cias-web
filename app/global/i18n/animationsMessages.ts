import { defineMessages } from 'react-intl';

import { NarratorAnimation } from 'models/Narrator';

export const scope = 'app.global.Animations';

export default defineMessages<NarratorAnimation>({
  uncertain: {
    id: `${scope}.uncertain`,
    defaultMessage: 'Uncertain',
  },
  wave: {
    id: `${scope}.wave`,
    defaultMessage: 'Wave',
  },
  greet: {
    id: `${scope}.greet`,
    defaultMessage: 'Greet',
  },
  otohLeftHand: {
    id: `${scope}.otohLeftHand`,
    defaultMessage: 'On the other hand (left)',
  },
  otohRightHand: {
    id: `${scope}.otohRightHand`,
    defaultMessage: 'On the other hand (right)',
  },
  pointRight: {
    id: `${scope}.pointRight`,
    defaultMessage: 'Point right',
  },
  pointLeft: {
    id: `${scope}.pointLeft`,
    defaultMessage: 'Point left',
  },
  pointUp: {
    id: `${scope}.pointUp`,
    defaultMessage: 'Point up',
  },
  pointDown: {
    id: `${scope}.pointDown`,
    defaultMessage: 'Point down',
  },
  pointLeftUp: {
    id: `${scope}.pointLeftUp`,
    defaultMessage: 'Point left up',
  },
  pointRightUp: {
    id: `${scope}.pointRightUp`,
    defaultMessage: 'Point right up',
  },
  explain: {
    id: `${scope}.explain`,
    defaultMessage: 'Explain',
  },
  rest: {
    id: `${scope}.rest`,
    defaultMessage: 'Rest',
  },
  restShoulderRubbing: {
    id: `${scope}.restShoulderRubbing`,
    defaultMessage: 'Rest shoulder rubbing',
  },
  restStretching: {
    id: `${scope}.restStretching`,
    defaultMessage: 'Rest stretching',
  },
  restWeightShift: {
    id: `${scope}.restWeightShift`,
    defaultMessage: 'Rest weight shift',
  },
  read: {
    id: `${scope}.read`,
    defaultMessage: 'Read',
  },
  reading: {
    id: `${scope}.reading`,
    defaultMessage: 'Reading',
  },
  confused: {
    id: `${scope}.confused`,
    defaultMessage: 'Confused',
  },
  doMagic: {
    id: `${scope}.doMagic`,
    defaultMessage: 'Do magic',
  },
  getAttention: {
    id: `${scope}.getAttention`,
    defaultMessage: 'Get attention',
  },
  search: {
    id: `${scope}.search`,
    defaultMessage: 'Search',
  },
  suggest: {
    id: `${scope}.suggest`,
    defaultMessage: 'Suggest',
  },
  surprised: {
    id: `${scope}.surprised`,
    defaultMessage: 'Surprised',
  },
  congratulate: {
    id: `${scope}.congratulate`,
    defaultMessage: 'Congratulate',
  },
  flyIn: {
    id: `${scope}.flyIn`,
    defaultMessage: 'Fly in',
  },
  flyOut: {
    id: `${scope}.flyOut`,
    defaultMessage: 'Fly out',
  },
  standStill: {
    id: `${scope}.standStill`,
    defaultMessage: 'Stand still',
  },
  process: {
    id: `${scope}.process`,
    defaultMessage: 'Process',
  },
  listen: {
    id: `${scope}.listen`,
    defaultMessage: 'Listen',
  },
  think: {
    id: `${scope}.think`,
    defaultMessage: 'Think',
  },
  write: {
    id: `${scope}.write`,
    defaultMessage: 'Write',
  },
  idle: {
    id: `${scope}.idle`,
    defaultMessage: 'Idle',
  },
  turnPage: {
    id: `${scope}.turnPage`,
    defaultMessage: 'Turn page',
  },
  openBook: {
    id: `${scope}.openBook`,
    defaultMessage: 'Open book',
  },
  closeBook: {
    id: `${scope}.closeBook`,
    defaultMessage: 'Close book',
  },
  announce: {
    id: `${scope}.announce`,
    defaultMessage: 'Announce',
  },
  blink: {
    id: `${scope}.blink`,
    defaultMessage: 'Blink',
  },
  acknowledge: {
    id: `${scope}.acknowledge`,
    defaultMessage: 'Acknowledge',
  },
  acknowledgeThoughtful: {
    id: `${scope}.acknowledgeThoughtful`,
    defaultMessage: 'Acknowledge thoughtful',
  },
  browsUp: {
    id: `${scope}.browsUp`,
    defaultMessage: 'Brows up',
  },
  decline: {
    id: `${scope}.decline`,
    defaultMessage: 'Decline',
  },
  declineAnnoyed: {
    id: `${scope}.declineAnnoyed`,
    defaultMessage: 'Decline annoyed',
  },
  declineThoughtful: {
    id: `${scope}.declineThoughtful`,
    defaultMessage: 'Decline thoughtful',
  },
  pleased: {
    id: `${scope}.pleased`,
    defaultMessage: 'Pleased',
  },
  sad: {
    id: `${scope}.sad`,
    defaultMessage: 'Sad',
  },
  hearLeftEar: {
    id: `${scope}.hearLeftEar`,
    defaultMessage: 'Hear left ear',
  },
  hearRightEar: {
    id: `${scope}.hearRightEar`,
    defaultMessage: 'Hear right ear',
  },
  hearBothEars: {
    id: `${scope}.hearBothEars`,
    defaultMessage: 'Hear both ears',
  },
  yawn: {
    id: `${scope}.yawn`,
    defaultMessage: 'Yawn',
  },
  eatCracker: {
    id: `${scope}.eatCracker`,
    defaultMessage: 'Eat cracker',
  },
  wearSunglasses: {
    id: `${scope}.wearSunglasses`,
    defaultMessage: 'Wear sunglasses',
  },
  lookDown: {
    id: `${scope}.lookDown`,
    defaultMessage: 'Look down',
  },
  lookDownAndBlink: {
    id: `${scope}.lookDownAndBlink`,
    defaultMessage: 'Look down and blink',
  },
  glanceUp: {
    id: `${scope}.glanceUp`,
    defaultMessage: 'Glance up',
  },
  glanceDown: {
    id: `${scope}.glanceDown`,
    defaultMessage: 'Glance down',
  },
  glanceLeft: {
    id: `${scope}.glanceLeft`,
    defaultMessage: 'Glance left',
  },
  glanceRight: {
    id: `${scope}.glanceRight`,
    defaultMessage: 'Glance right',
  },
  moveLeft: {
    id: `${scope}.moveLeft`,
    defaultMessage: 'Move left',
  },
  moveRight: {
    id: `${scope}.moveRight`,
    defaultMessage: 'Move right',
  },
});
