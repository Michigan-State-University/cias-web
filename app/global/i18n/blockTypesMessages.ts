import { defineMessages } from 'react-intl';

export const scope = 'app.global.BlockTypes';

export default defineMessages({
  BodyAnimation: {
    id: `${scope}.BodyAnimation`,
    defaultMessage: 'Body Animation',
  },
  Speech: {
    id: `${scope}.Speech`,
    defaultMessage: 'Speech',
  },
  Reflection: {
    id: `${scope}.Reflection`,
    defaultMessage: 'Speech',
  },
  ReflectionFormula: {
    id: `${scope}.ReflectionFormula`,
    defaultMessage: 'Speech',
  },
  HeadAnimation: {
    id: `${scope}.HeadAnimation`,
    defaultMessage: 'Head Animation',
  },
  ReadQuestion: {
    id: `${scope}.ReadQuestion`,
    defaultMessage: 'Read question',
  },
  Pause: {
    id: `${scope}.Pause`,
    defaultMessage: 'Pause',
  },
  Feedback: {
    id: `${scope}.Feedback`,
    defaultMessage: 'Show Spectrum',
  },
});
