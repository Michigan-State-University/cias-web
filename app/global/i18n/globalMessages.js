import { defineMessages } from 'react-intl';

export const scope = 'app.GlobalMessages';

export default defineMessages({
  variables: {
    variableNamePlaceholder: {
      id: `${scope}.variableNamePlaceholder`,
      defaultMessage: 'Variable name...',
    },
    variableScorePlaceholder: {
      id: `${scope}.variableScorePlaceholder`,
      defaultMessage: 'Score',
    },
    value: {
      id: `${scope}.value`,
      defaultMessage: 'Value:',
    },
    emptyBadge: {
      id: `${scope}.emptyBadge`,
      defaultMessage: 'No assigned value',
    },
  },
  blockTypes: {
    BodyAnimation: {
      id: `${scope}.BodyAnimation`,
      defaultMessage: 'Body Animation',
    },
    Speech: {
      id: `${scope}.Speech`,
      defaultMessage: 'Speech',
    },
    Reflection: {
      id: `${scope}.Reflection`,
      defaultMessage: 'Speech',
    },
    HeadAnimation: {
      id: `${scope}.HeadAnimation`,
      defaultMessage: 'Head Animation',
    },
    ReadQuestion: {
      id: `${scope}.ReadQuestion`,
      defaultMessage: 'Read question',
    },
  },
  questionTypes: {
    'Question::Single': {
      id: `${scope}.Question::Single`,
      defaultMessage: 'Single answer screen',
    },
    'Question::Multiple': {
      id: `${scope}.Question::Multiple`,
      defaultMessage: 'Multi answer screen',
    },
    'Question::TextBox': {
      id: `${scope}.Question::TextBox`,
      defaultMessage: 'Textbox screen',
    },
    'Question::Number': {
      id: `${scope}.Question::Number`,
      defaultMessage: 'Number screen',
    },
    'Question::Grid': {
      id: `${scope}.Question::Grid`,
      defaultMessage: 'Grid screen',
    },
    'Question::AnalogueScale': {
      id: `${scope}.Question::AnalogueScale`,
      defaultMessage: 'Visual Analogue Scale screen',
    },
    'Question::Information': {
      id: `${scope}.Question::Information`,
      defaultMessage: 'Information screen',
    },
    'Question::Url': {
      id: `${scope}.Question::Url`,
      defaultMessage: 'Url screen',
    },
    'Question::FollowUpContact': {
      id: `${scope}.Question::FollowUpContact`,
      defaultMessage: 'FollowUpContact screen',
    },
  },
  statuses: {
    draft: {
      id: `${scope}.draft`,
      defaultMessage: 'Draft',
    },
    published: {
      id: `${scope}.published`,
      defaultMessage: 'Published',
    },
    closed: {
      id: `${scope}.closed`,
      defaultMessage: 'Closed',
    },
    archived: {
      id: `${scope}.archived`,
      defaultMessage: 'Archived',
    },
  },
  createProblemError: {
    id: `${scope}.createProblemError`,
    defaultMessage: `Couldn't create an intervention`,
  },
  editProblemError: {
    id: `${scope}.editProblemError`,
    defaultMessage: `Couldn't edit an intervention`,
  },
});
