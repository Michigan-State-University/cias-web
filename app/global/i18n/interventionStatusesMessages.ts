import { defineMessages } from 'react-intl';

export const scope = 'app.global.InterventionStatusesMessages';

export default defineMessages({
  draft: {
    id: `${scope}.draft`,
    defaultMessage: 'Draft',
  },
  published: {
    id: `${scope}.published`,
    defaultMessage: 'Published',
  },
  closed: {
    id: `${scope}.closed`,
    defaultMessage: 'Closed',
  },
  archived: {
    id: `${scope}.archived`,
    defaultMessage: 'Archived',
  },
});
