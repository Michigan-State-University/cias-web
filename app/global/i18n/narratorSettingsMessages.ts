import { defineMessages } from 'react-intl';

import { NarratorSettingsKey } from 'models/Narrator';

export const scope = 'app.global.NarratorSettings';

export default defineMessages<NarratorSettingsKey>({
  [NarratorSettingsKey.VOICE]: {
    id: `${scope}.voice`,
    defaultMessage: 'Voice',
  },
  [NarratorSettingsKey.ANIMATION]: {
    id: `${scope}.animation`,
    defaultMessage: 'Display Narrator',
  },
  [NarratorSettingsKey.EXTRA_SPACE_FOR_NARRATOR]: {
    id: `${scope}.extra_space_for_narrator`,
    defaultMessage: 'Extra Space for Narrator',
  },
  [NarratorSettingsKey.CHARACTER]: {
    id: `${scope}.character`,
    defaultMessage: 'Character',
  },
});
