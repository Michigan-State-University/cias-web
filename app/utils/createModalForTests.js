import ReactDOM from 'react-dom';

const createModalForTests = () => {
  ReactDOM.createPortal = jest.fn(element => element);

  const modalContainer = document.createElement('div');
  modalContainer.setAttribute('id', 'modal-portal');
  document.body.appendChild(modalContainer);

  const mainAppContainer = document.createElement('div');
  mainAppContainer.setAttribute('id', 'main-app-container');
  document.body.appendChild(mainAppContainer);
};

export default createModalForTests;
