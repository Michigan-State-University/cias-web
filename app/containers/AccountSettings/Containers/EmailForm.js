import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import { compose } from 'redux';
import { Formik } from 'formik';
import { createStructuredSelector } from 'reselect';
import * as Yup from 'yup';

import Button from 'components/Button';
import Column from 'components/Column';
import FormikInput from 'components/FormikInput';
import H3 from 'components/H3';
import Modal from 'components/Modal';
import Row from 'components/Row';
import { useInjectSaga } from 'utils/injectSaga';
import {
  editUserRequest,
  editUserSaga,
  makeSelectUser,
} from 'global/reducers/auth';

import messages from '../messages';

const validationSchema = formatMessage =>
  Yup.object().shape({
    email: Yup.string().required(formatMessage(messages.emailRequired)),
    passwordConfirmation: Yup.string().required(
      formatMessage(messages.oldPasswordRequired),
    ),
  });

const initialValues = user => ({
  email: user.email,
  passwordConfirmation: '',
});

const EmailForm = ({ formatMessage, user }) => {
  const [modalVisible, setModalVisible] = useState(false);
  const openModal = () => setModalVisible(true);
  const closeModal = () => setModalVisible(false);

  useInjectSaga({ key: 'editUser', saga: editUserSaga });

  // eslint-disable-next-line no-unused-vars
  const onSubmit = ({ email, passwordConfirmation }, { setSubmitting }) => {
    setSubmitting(false);
  };

  const handleBlur = email => () => {
    if (user.email !== email) openModal();
  };

  return (
    <Fragment>
      <Formik
        validationSchema={validationSchema(formatMessage)}
        initialValues={initialValues(user)}
        onSubmit={onSubmit}
      >
        {({ handleSubmit, values }) => {
          const inputProps = {
            height: 46,
            width: '100%',
          };
          return (
            <Fragment>
              <Modal
                visible={modalVisible}
                title={formatMessage(messages.changeYourEmail)}
                onClose={closeModal}
                maxWidth={500}
              >
                <Column>
                  <H3 mb={30}>
                    <FormattedMessage {...messages.modalDescriptipn} />
                  </H3>
                  <FormikInput
                    formikKey="passwordConfirmation"
                    placeholder={formatMessage(messages.password)}
                    label={formatMessage(messages.passwordLabel)}
                    type="password"
                    inputProps={inputProps}
                  />
                  <Row width="100%" mt={30}>
                    <Button
                      mr={20}
                      inverted
                      hoverable
                      onClick={closeModal}
                      type="button"
                    >
                      <FormattedMessage {...messages.cancel} />
                    </Button>
                    <Button hoverable onClick={handleSubmit} type="button">
                      <FormattedMessage {...messages.changeEmail} />
                    </Button>
                  </Row>
                </Column>
              </Modal>
              <Row width="100%">
                <FormikInput
                  formikKey="email"
                  placeholder={formatMessage(messages.email)}
                  label={formatMessage(messages.emailLabel)}
                  type="text"
                  inputProps={{
                    onBlur: handleBlur(values.email),
                    ...inputProps,
                  }}
                />
              </Row>
            </Fragment>
          );
        }}
      </Formik>
    </Fragment>
  );
};

EmailForm.propTypes = {
  formatMessage: PropTypes.func,
  user: PropTypes.shape({
    email: PropTypes.string,
  }),
};

const mapStateToProps = createStructuredSelector({
  user: makeSelectUser(),
});

const mapDispatchToProps = {
  editUser: editUserRequest,
};

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(EmailForm);
