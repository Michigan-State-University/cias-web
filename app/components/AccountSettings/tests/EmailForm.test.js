/**
 *
 * Tests for AccountSettings
 *
 */

import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import 'jest-styled-components';
import { DEFAULT_LOCALE } from 'i18n';
import { IntlProvider } from 'react-intl';
import createModalForTests from 'utils/createModalForTests';

import EmailForm from '../EmailForm';

describe('<EmailForm />', () => {
  beforeAll(() => {
    createModalForTests();
  });
  const defaultProps = {
    formatMessage: jest.fn(msg => msg.defaultMessage),
    user: { email: 'test@test.pl' },
    changeEmail: jest.fn(),
    error: '',
    loading: false,
    changeErrorValue: jest.fn(),
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <EmailForm {...defaultProps} />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <EmailForm {...defaultProps} />
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should render error and match the snapshot', () => {
    const { container } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <EmailForm {...defaultProps} error="Test error" />
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should call edit email correctly', async () => {
    const { getByTestId } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <EmailForm {...defaultProps} />
      </IntlProvider>,
    );

    const newEmail = 'new@test.com';
    const oldPassword = 'Password1!';
    const emailInput = getByTestId('email-input').querySelector('input');
    fireEvent.change(emailInput, { target: { value: newEmail } });
    fireEvent.blur(emailInput);
    const passwordInput = getByTestId('password-confirmation').querySelector(
      'input',
    );
    fireEvent.change(passwordInput, { target: { value: oldPassword } });
    fireEvent.blur(passwordInput);
    const confirmButton = getByTestId('confirm-button');
    fireEvent.click(confirmButton);

    await waitFor(() => {
      expect(defaultProps.changeEmail).toHaveBeenCalledWith({
        newEmail,
        oldPassword,
      });
    });
  });

  it('Should call close modal', async () => {
    const { getByTestId } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <EmailForm {...defaultProps} />
      </IntlProvider>,
    );

    const newEmail = 'new@test.com';
    const emailInput = getByTestId('email-input').querySelector('input');
    fireEvent.change(emailInput, { target: { value: newEmail } });
    fireEvent.blur(emailInput);
    const closeButton = getByTestId('close-button');
    fireEvent.click(closeButton);

    await waitFor(() => {
      expect(defaultProps.changeErrorValue).toHaveBeenCalled();
    });
  });
});
