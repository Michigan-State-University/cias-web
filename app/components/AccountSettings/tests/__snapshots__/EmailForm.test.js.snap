// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailForm /> Should render and match the snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c4 {
  padding: 12px;
  border-style: solid;
  border-width: 1px;
  border-color: #E9F1FB;
  border-radius: 5px;
  width: 100%;
  height: 46px;
  color: #2F3850;
}

.c4:focus {
  box-shadow: none;
  outline: none;
  border-color: #107969;
}

.c4:disabled {
  color: #ABBBD1;
}

.c2 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-bottom: 5px;
  color: #2F3850;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

<div>
  <div
    class="c0"
    width="100%"
  >
    <div
      class="c1"
      data-testid="email-input"
    >
      <p
        class="c2"
        width="fit-content"
      >
        E-mail address
      </p>
      <div
        class="c3"
        width="100%"
      >
        <input
          class="c4"
          height="46"
          name="email"
          placeholder="Enter e-mail address"
          type="text"
          value="test@test.pl"
          width="100%"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<EmailForm /> Should render error and match the snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c4 {
  padding: 12px;
  border-style: solid;
  border-width: 1px;
  border-color: #E9F1FB;
  border-radius: 5px;
  width: 100%;
  height: 46px;
  color: #2F3850;
}

.c4:focus {
  box-shadow: none;
  outline: none;
  border-color: #107969;
}

.c4:disabled {
  color: #ABBBD1;
}

.c2 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-bottom: 5px;
  color: #2F3850;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

<div>
  <div
    class="c0"
    width="100%"
  >
    <div
      class="c1"
      data-testid="email-input"
    >
      <p
        class="c2"
        width="fit-content"
      >
        E-mail address
      </p>
      <div
        class="c3"
        width="100%"
      >
        <input
          class="c4"
          height="46"
          name="email"
          placeholder="Enter e-mail address"
          type="text"
          value="test@test.pl"
          width="100%"
        />
      </div>
    </div>
  </div>
</div>
`;
